x-pack/plugins/security/server/authentication/authenticator.ts

```diff
@@ -946,3 +952,37 @@ export class Authenticator {
       : `${this.options.basePath.serverBasePath}/security/logged_out?${searchParams.toString()}`;
   }
 }
+
+function enrichWithUserProfileId(
```

Comment:
__concern:__ since we updated authenticated user mock to always populate `profile_uid` we might not really test this method in `Authenticator` unit tests. Would you mind adding a few tests for login/authenticate/re-authenticate that explicitly test this logic (e.g. create user mock returned by the provider _without_ `profile_uid`)?

Reply:
Good point, I've added unit tests for this method.